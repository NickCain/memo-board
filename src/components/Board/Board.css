.Board {
    display: flex;
    flex-flow: row wrap;
}
