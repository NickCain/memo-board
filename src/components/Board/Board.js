import PropTypes from 'prop-types';
import React, { Component } from 'react';

import Idea from '../Idea/Idea';

import './Board.css';

class Board extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { ideasList, onUpdateIdea } = this.props;
    return (
      <div className="Board">
        {
          ideasList.map(idea => (
            <Idea
              key={idea.id}
              id={idea.id}
              createdDate={idea.created_date}
              onUpdateIdea={onUpdateIdea}
              body={idea.body || 'text'}
              title={idea.title  || 'title'}
            />
          ))
        }
      </div>
    );
  }
}

Board.propTypes = {
  onUpdateIdea: PropTypes.func,
  ideasList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      created_date: PropTypes.number
    })
  )
};

export default Board;

