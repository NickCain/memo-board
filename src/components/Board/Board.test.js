import { mount, shallow } from 'enzyme';
import React from 'react';
import Board from './Board.js'

let wrapper;
let shallowWrapper;

describe('Board - Container', () => {
  describe('Board renders', () => {
    beforeEach(() => {
      wrapper = mount(
        <Board />
      );
      shallowWrapper = shallow(
        <Board />
      );
    });

    it('should render the Board component', () => {
      expect(shallowWrapper).toMatchSnapshot();
      expect(wrapper.length).toBeGreaterThan(0);
    });
  });
});
