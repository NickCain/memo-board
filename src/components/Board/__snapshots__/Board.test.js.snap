// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board - Container Board renders should render the Board component 1`] = `undefined`;
